contracts/periphery/contracts/s2s/SmartChainXLib.sol

```diff
@@ -118,9 +166,13 @@ library SmartChainXLib {
         );
 
         // 2. Decode `OutboundLaneData` and return the latest nonce
-        CommonTypes.OutboundLaneData memory outboundLaneData = CommonTypes
-            .decodeOutboundLaneData(data);
-        return outboundLaneData.latestGeneratedNonce;
+        if (data.length == 0) {
+            return 0;
```

Comment:
`return 0` means that the latest generated nonce is 0. Just revert it may be better.

Reply:
right